Sources/AsyncHTTPClient/NIOTransportServices/TLSConfiguration.swift

```diff
@@ -0,0 +1,127 @@
+internal extension TLSConfiguration {
+    
+    /// Dispatch queue used by Network framework TLS to control certificate verification
+    static var tlsDispatchQueue = DispatchQueue(label: "TLSDispatch")
+
+    /// create NWProtocolTLS.Options for use with NIOTransportServices from the NIOSSL TLSConfiguration
+    ///
+    /// - Parameter queue: Dispatch queue to run `sec_protocol_options_set_verify_block` on.
+    /// - Returns: Equivalent NWProtocolTLS Options
+    func getNWProtocolTLSOptions() -> NWProtocolTLS.Options {
+        let options = NWProtocolTLS.Options()
+        
+        // minimum TLS protocol
+        if #available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *) {
+            sec_protocol_options_set_min_tls_protocol_version(options.securityProtocolOptions, self.minimumTLSVersion.nwTLSProtocolVersion)
+        }
+        
+        // maximum TLS protocol
+        if let maximumTLSVersion = self.maximumTLSVersion {
+            if #available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *) {
+                sec_protocol_options_set_max_tls_protocol_version(options.securityProtocolOptions, maximumTLSVersion.nwTLSProtocolVersion)
+            } else {
+                precondition(self.maximumTLSVersion != nil, "TLSConfiguration.maximumTLSVersion is not supported")
+            }
+        }
+
+        // application protocols
+        if self.applicationProtocols.count > 0 {
+            preconditionFailure("TLSConfiguration.applicationProtocols is not supported")
+        }
+        /*for applicationProtocol in self.applicationProtocols {
+            applicationProtocol.utf8.withContiguousStorageIfAvailable { buffer in
+                guard let opaquePointer = OpaquePointer(buffer.baseAddress) else { return }
+                let int8Pointer = UnsafePointer<Int8>(opaquePointer)
+                sec_protocol_options_add_tls_application_protocol(options.securityProtocolOptions, int8Pointer)
+            }
+        }*/
+
+        // the certificate chain
+        if self.certificateChain.count > 0 {
+            preconditionFailure("TLSConfiguration.certificateChain is not supported")
+        }
+        
+        // cipher suites
+        if self.cipherSuites.count > 0 {
+            //preconditionFailure("TLSConfiguration.cipherSuites is not supported")
+        }
+        
+        // key log callback
+        if let _ = self.keyLogCallback {
+            preconditionFailure("TLSConfiguration.keyLogCallback is not supported")
+        }
+        
+        // private key
+        if let _ = self.privateKey {
+            preconditionFailure("TLSConfiguration.privateKey is not supported")
+        }
+        
+        // renegotiation support key is unsupported
+        
+        // trust roots
+        if let trustRoots = self.trustRoots {
+            guard case .default = trustRoots else {
+                preconditionFailure("TLSConfiguration.trustRoots != .default is not supported")
+            }
```

Comment:
This one is also reasonably do-able, if we configure a custom trust challenge. For now, though, let's leave it as-is.

Reply:
I assume this would be done with `sec_protocol_options_set_verify_block`. There doesn't seem to be any documentation on what the default verify block would involve, outside of evaluating the trust (Does it also verify server address?). So wasn't keen on setting one up outside the very specific case of having no certificate verification.

Reply:
That's correct. You start out with a trust that's appropriately configured (it's passed in, I believe) and you can modify it as needed.